base/windows/shop/InventoryWindow.js

```diff
@@ -0,0 +1,338 @@
+                }
+
+                if (this.item_grid[line][col].equipped) {
+                    let dead_icons = this.icon_group.children.filter(e => { return (e.alive === false && e.text === undefined); });
+                    if(dead_icons.length>0) dead_icons[0].reset(col*ICON_SIZE, line*ICON_SIZE);
+                    else this.window.create_at_group(col*ICON_SIZE, line*ICON_SIZE, "equipped", undefined,undefined, "icons");
+                }
+
+                if (this.item_grid[line][col].quantity > 1) {
+                    let dead_text = this.icon_group.children.filter(t => { return (t.alive === false && t.text !== undefined); });
+                    if(dead_text.length>0){
+                        dead_text[0].text = this.item_grid[line][col].quantity.toString();
+                        dead_text[0].reset(col*ICON_SIZE, line*ICON_SIZE);
+                        dead_text[0].x += (SUB_TEXT_X_SHIFT - dead_text[0].width);
+                    }
+                    else{
+                        let item_count = this.game.add.bitmapText(col*ICON_SIZE, line*ICON_SIZE, 'gs-item-bmp-font', this.item_grid[line][col].quantity.toString());
+                        item_count.x += (SUB_TEXT_X_SHIFT - item_count.width);
+                        this.window.add_to_internal_group("icons", item_count);
+                    }
+                }
+            }
+            if(finish) break;
+        }
+        this.sprite_group.alpha = 1;
+    }
+
+    /*Opens this window for a given character
+
+    Input: char_key [string] - The character's key name
+           item [string] - The item to check against
+           expand [boolean] - If true, the window will be in expanded state
+           close_callback [function] - Callback function (Optional)
+           open_callback [function] - Callback function (Optional)*/
+    open(char_key, item=undefined, expand=false, close_callback, open_callback){
+        this.char = this.data.info.party_data.members.filter(c => { return (c.key_name === char_key)})[0];
+        this.selected_item = item;
+
+        this.make_item_grid();
+        this.check_expand(expand);
+        this.set_sprites();
+
+        this.is_open = true;
+        this.close_callback = close_callback;
+        this.window.show(open_callback, false);
+    }
+
+    /*Clears information and closes the window
+
+    Input: destroy [boolean] - If true, sprites are destroyed*/
+    close(destroy = false){
+        kill_all_sprites(this.sprite_group, destroy);
+        kill_all_sprites(this.icon_group, destroy);
+
+        this.text.text.alpha = 0;
+        this.text.shadow.alpha = 0;
+        this.char = null;
+        this.selected_item = null;
+        this.cursor_pos = {line: 0, col: 0};
+        this.item_grid = [];
+        this.check_expand(false);
+
+        this.is_open = false;
+        this.window.close(this.close_callback, false);
+        this.close_callback = null;
+    }
+}
```

Comment:
There are many key names in string literals in this file. Please consider using variables.